clients/go/zbc/envWrapper.go

```diff
@@ -0,0 +1,76 @@
+// Copyright © 2018 Camunda Services GmbH (info@camunda.com)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package zbc
+
+import (
+	"os"
+)
+
+var systemEnv = newEnvironmentWrapper()
+
+type environment interface {
+	getenv(variable string) string
+	setenv(variable, value string)
+	unsetenv(variable string)
+	lookupEnv(variable string) (string, bool)
+	environCopy() map[string]string
+	overwrite(environ map[string]string)
+}
+
+type envWrapper struct {
+	vars map[string]string
```

Comment:
One thing to note, in Go maps cannot be concurrently accessed even for reading I think (could be wrong, but I think we ran into this issue with the cache). Could this be an issue here? See the YAML cache impl